lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -25,19 +25,34 @@ using namespace mlir;
 // SVModuleOp
 //===----------------------------------------------------------------------===//
 
+void SVModuleOp::build(mlir::OpBuilder &builder, mlir::OperationState &state,
+                       llvm::StringRef name, hw::ModuleType type) {
+  state.addAttribute(SymbolTable::getSymbolAttrName(),
+                     builder.getStringAttr(name));
+  state.addAttribute(getModuleTypeAttrName(state.name), TypeAttr::get(type));
+  state.addRegion();
+}
+
 void SVModuleOp::print(OpAsmPrinter &p) {
   p << " ";
+
+  // Print the visibility of the module.
+  StringRef visibilityAttrName = SymbolTable::getVisibilityAttrName();
+  if (auto visibility = (*this)->getAttrOfType<StringAttr>(visibilityAttrName))
+    p << visibility.getValue() << ' ';
+
   p.printSymbolName(SymbolTable::getSymbolName(*this).getValue());
   hw::module_like_impl::printModuleSignatureNew(p, getBodyRegion(),
                                                 getModuleType(), {}, {});
-  p.printOptionalAttrDictWithKeyword(getOperation()->getAttrs(),
-                                     getAttributeNames());
```

Comment:
We should keep this.

Reply:
I think I misunderstood your previous comments. It was already been reverted and applied the new way.